src/services/query.ts

```diff
@@ -14,26 +14,19 @@ import Pino from 'pino';
 import PrettyMilliseconds from 'pretty-ms';
 import * as RDF from 'rdf-js';
 import {Term, TermsTransformer} from './terms';
-import {Distribution} from '@netwerk-digitaal-erfgoed/network-of-terms-catalog';
+import {
+  Distribution,
+  IRI,
+} from '@netwerk-digitaal-erfgoed/network-of-terms-catalog';
 
 export interface ConstructorOptions {
   logger: Pino.Logger;
-  distribution: Distribution;
-  query: string;
   comunica: IActorInitSparqlArgs;
-  timeoutMs: number;
 }
 
 const schemaConstructor = Joi.object({
   logger: Joi.object().required(),
-  distribution: Joi.object().required(),
-  query: Joi.string().required(),
   comunica: Joi.object().required(),
-  timeoutMs: Joi.number()
-    .integer()
-    .min(1)
-    .max(parseInt(process.env.MAX_QUERY_TIMEOUT as string) || 10000)
-    .default(parseInt(process.env.DEFAULT_QUERY_TIMEOUT as string) || 5000),
```

Comment:
Parameterised this arguments as part of the `lookup()` and `search()` functions instead.

Reply:
👍🏻